Lib/ntpath.py

```diff
@@ -554,9 +567,20 @@ def _getfinalpathname_nonstrict(path):
         except OSError:
             pass
 
-        # Allow file (2) or directory (3) not found, incorrect parameter (87),
-        # invalid syntax (123), and symlinks that cannot be followed (1921)
-        allowed_winerror = 2, 3, 87, 123, 1921
+        # These error codes indicate that we should stop resolving the path
+        # and return the value we currently have.
+        # 1: ERROR_INVALID_FUNCTION
+        # 2: ERROR_FILE_NOT_FOUND
+        # 3: ERROR_DIRECTORY_NOT_FOUND
+        # 5: ERROR_ACCESS_DENIED
+        # 21: ERROR_NOT_READY (implies drive with no media)
+        # 32: ERROR_SHARING_VIOLATION
+        # 50: ERROR_NOT_SUPPORTED (implies no file system)
```

Comment:
Actually, for a device that's not a file system or not mounted by a file system, the error is typically either `ERROR_INVALID_FUNCTION` (e.g. from `STATUS_INVALID_DEVICE_REQUEST` for the file-system control `NtFsControlFile`: `FSCTL_GET_REPARSE_POINT`) or `ERROR_INVALID_PARAMETER` (e.g. from `STATUS_INVALID_INFO_CLASS` for the file-system query  `NtQueryInformationFile`: `FileNameInformation`). 

Getting `ERROR_NOT_SUPPORTED` (from `STATUS_NOT_SUPPORTED`) is usually for a specific request that's not supported, as opposed to a class of request that's not supported. For example, the named-pipe file system doesn't support `FSCTL_GET_REPARSE_POINT`.  A driver may also return `STATUS_NOT_IMPLEMENTED` in this case, but that maps to `ERROR_INVALID_FUNCTION`.